### PR TITLE
Upgrade to RxJava3

### PR DESCRIPTION
**Pull request by BraisGabin**
I changed the package name, the coordinates and the version. Tell me if that's OK or I should keep some of those.

I did some minor changes in the tests too because the test API changed a lot between 2.x and 3.x.

This PR is related with #55.

### CODE REVIEW

**Comment by BraisGabin** [comment]
Rebased

**Comment by BraisGabin** [comment]
I have a commit ready to update the code from Java 6 to Java 8 but it has conflicts with this branch so I'll wait to this to be merged.

**Review by ZacSweers — APPROVED** [review]
My migrator script + manual fixes in our codebase arrived at basically the same changes and are working fine for us. This looks good to me :+1:

File: pom.xml
Comment (ZacSweers) [inline pom.xml]: worth picking up the latest patch

```suggestion
      <version>3.0.2</version>
```

File: pom.xml
Comment (lennynilsson) [inline pom.xml:14]: Current version is 3.0.3. @BraisGabin would you be so kind? :)

**Comment by valeriyo** [comment]
Thanks @BraisGabin for doing the work! Is it waiting on @JakeWharton 's approval?

File: src/test/java/com/jakewharton/rxrelay3/SerializedRelayTest.java
Comment (JakeWharton) [inline src/test/java/com/jakewharton/rxrelay3/SerializedRelayTest.java:176]: Why doesn't this use normal equals?

Comment (BraisGabin) [reply src/test/java/com/jakewharton/rxrelay3/SerializedRelayTest.java:176]: This is because the code didn't ensure the position and I didn't want to change the behaviour. But now that I check it with `actual.size() == expected.size() && actual.containsAll(expected)` is enough.

**Pull request merged**
